indico/modules/events/sessions/models/sessions.py

```diff
@@ -152,6 +154,28 @@ def session(self):
         """Convenience property so all event entities have it"""
         return self
 
+    @property
+    def start_dt(self):
+        from indico.modules.events.sessions.models.blocks import SessionBlock
+        start_dt = (self.event_new.timetable_entries
+                    .with_entities(TimetableEntry.start_dt)
+                    .join('session_block')
+                    .filter(TimetableEntry.type == TimetableEntryType.SESSION_BLOCK, SessionBlock.session == self)
+                    .first())
+        if start_dt:
+            return start_dt[0]
+
+    @property
+    def end_dt(self):
+        from indico.modules.events.sessions.models.blocks import SessionBlock
+        return (SessionBlock.query.with_parent(self)
+                .join(TimetableEntry)
+                .options(noload('*'))
+                .options(load_only('id'))
+                .options(contains_eager('timetable_entry').load_only('start_dt'))
+                .order_by((TimetableEntry.start_dt + SessionBlock.duration).desc())
+                .first().timetable_entry.end_dt)
```

Comment:
Did you try this if the session has no blocks? `.timetable_entry.end_dt` might be called on a `None` value...